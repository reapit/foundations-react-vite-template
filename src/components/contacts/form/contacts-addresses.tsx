import { cx } from '@linaria/core'
import {
  BodyText,
  InputWrap,
  InputGroup,
  Label,
  Select,
  FormLayout,
  elFadeIn,
  elMb11,
  InputError,
  Button,
  elMb7,
} from '@reapit/elements'
import { Dispatch, FC, SetStateAction, useState } from 'react'
import { UseFormReturn } from 'react-hook-form'
import { COUNTRY_OPTIONS } from './contacts-country-options'
import { ContactFormSchema } from './contacts-validation-schema'

export const ADDRESS_TYPE_OPTIONS = [
  { name: 'Primary', value: 'primary' },
  { name: 'Secondary', value: 'secondary' },
  { name: 'Home', value: 'home' },
  { name: 'Work', value: 'work' },
  { name: 'Forwarding', value: 'forwarding' },
  { name: 'Company', value: 'company' },
  { name: 'Previous', value: 'previous' },
]

interface ContactsAddressesProps {
  form: UseFormReturn<ContactFormSchema, any>
}

export const handleAddAddress = (setHasAsAddress: Dispatch<SetStateAction<boolean>>, hasAddress: boolean) => () => {
  setHasAsAddress(!hasAddress)
}

export const ContactsAddresses: FC<ContactsAddressesProps> = ({ form }) => {
  const [hasSecondaryAddress, setHasSecondaryAddress] = useState<boolean>(false)
  const [hasWorkAddress, setHasWorkAddress] = useState<boolean>(false)

  const {
    register,
    formState: { errors },
  } = form

  return (
    <>
      <div className={elMb11}>
        <BodyText hasGreyText hasNoMargin>
          Provide up to three addresses for your contact. All addresses are optional, only primary address is shown by
          default.
        </BodyText>
      </div>
      <BodyText>Primary Address</BodyText>
      <FormLayout className={cx(elMb11, elFadeIn)}>
        <InputWrap>
          <InputGroup>
            <Select {...register('primaryAddress.type')} defaultValue="primary">
              <option key="default-option" value="">
                None selected
              </option>
              {ADDRESS_TYPE_OPTIONS.map(({ name, value }) => (
                <option key={value} value={value}>
                  {name}
                </option>
              ))}
            </Select>
            <Label>Address Type</Label>
          </InputGroup>
        </InputWrap>
        <InputWrap>
          <InputGroup
            {...register('primaryAddress.buildingName')}
            type="text"
            label="Building Name"
            placeholder="Name of the building"
          />
          {errors.primaryAddress?.buildingName?.message && (
            <InputError message={errors.primaryAddress.buildingName.message} />
          )}
        </InputWrap>
        <InputWrap>
          <InputGroup
            {...register('primaryAddress.buildingNumber')}
            type="text"
            label="Building Number"
            placeholder="Number of the building"
          />
          {errors.primaryAddress?.buildingNumber?.message && (
            <InputError message={errors.primaryAddress.buildingNumber.message} />
          )}
        </InputWrap>
        <InputWrap>
          <InputGroup {...register('primaryAddress.line1')} type="text" label="Line 1" placeholder="Address line 1" />
          {errors.primaryAddress?.line1?.message && <InputError message={errors.primaryAddress.line1.message} />}
        </InputWrap>
        <InputWrap>
          <InputGroup {...register('primaryAddress.line2')} type="text" label="Line 2" placeholder="Address line 2" />
          {errors.primaryAddress?.line2?.message && <InputError message={errors.primaryAddress.line2.message} />}
        </InputWrap>
        <InputWrap>
          <InputGroup {...register('primaryAddress.line3')} type="text" label="Line 3" placeholder="Address line 3" />
          {errors.primaryAddress?.line3?.message && <InputError message={errors.primaryAddress.line3.message} />}
        </InputWrap>
        <InputWrap>
          <InputGroup {...register('primaryAddress.line4')} type="text" label="Line 4" placeholder="Address line 4" />
          {errors.primaryAddress?.line4?.message && <InputError message={errors.primaryAddress.line4.message} />}
        </InputWrap>
        <InputWrap>
          <InputGroup
            {...register('primaryAddress.postcode')}
            type="text"
            label="Post Code"
            placeholder="Address post code"
          />
          {errors.primaryAddress?.postcode?.message && <InputError message={errors.primaryAddress.postcode.message} />}
        </InputWrap>
        <InputWrap>
          <InputGroup>
            <Select {...register('primaryAddress.countryId')} defaultValue="GB">
              <option key="default-option" value="">
                None selected
              </option>
              {COUNTRY_OPTIONS.map(({ name, value }) => (
                <option key={value} value={value}>
                  {name}
                </option>
              ))}
            </Select>
            <Label>Country</Label>
          </InputGroup>
        </InputWrap>
      </FormLayout>
      {hasSecondaryAddress ? (
        <>
          <BodyText>Secondary Address</BodyText>
          <FormLayout className={cx(elMb11, elFadeIn)}>
            <InputWrap>
              <InputGroup>
                <Select {...register('secondaryAddress.type')} defaultValue="secondary">
                  <option key="default-option" value="">
                    None selected
                  </option>
                  {ADDRESS_TYPE_OPTIONS.map(({ name, value }) => (
                    <option key={value} value={value}>
                      {name}
                    </option>
                  ))}
                </Select>
                <Label>Address Type</Label>
              </InputGroup>
            </InputWrap>
            <InputWrap>
              <InputGroup
                {...register('secondaryAddress.buildingName')}
                type="text"
                label="Building Name"
                placeholder="Name of the building"
              />
              {errors.secondaryAddress?.buildingName?.message && (
                <InputError message={errors.secondaryAddress.buildingName.message} />
              )}
            </InputWrap>
            <InputWrap>
              <InputGroup
                {...register('secondaryAddress.buildingNumber')}
                type="text"
                label="Building Number"
                placeholder="Number of the building"
              />
              {errors.secondaryAddress?.buildingNumber?.message && (
                <InputError message={errors.secondaryAddress.buildingNumber.message} />
              )}
            </InputWrap>
            <InputWrap>
              <InputGroup
                {...register('secondaryAddress.line1')}
                type="text"
                label="Line 1"
                placeholder="Address line 1"
              />
              {errors.secondaryAddress?.line1?.message && (
                <InputError message={errors.secondaryAddress.line1.message} />
              )}
            </InputWrap>
            <InputWrap>
              <InputGroup
                {...register('secondaryAddress.line2')}
                type="text"
                label="Line 2"
                placeholder="Address line 2"
              />
              {errors.secondaryAddress?.line2?.message && (
                <InputError message={errors.secondaryAddress.line2.message} />
              )}
            </InputWrap>
            <InputWrap>
              <InputGroup
                {...register('secondaryAddress.line3')}
                type="text"
                label="Line 3"
                placeholder="Address line 3"
              />
              {errors.secondaryAddress?.line3?.message && (
                <InputError message={errors.secondaryAddress.line3.message} />
              )}
            </InputWrap>
            <InputWrap>
              <InputGroup
                {...register('secondaryAddress.line4')}
                type="text"
                label="Line 4"
                placeholder="Address line 4"
              />
              {errors.secondaryAddress?.line4?.message && (
                <InputError message={errors.secondaryAddress.line4.message} />
              )}
            </InputWrap>
            <InputWrap>
              <InputGroup
                {...register('secondaryAddress.postcode')}
                type="text"
                label="Post Code"
                placeholder="Address post code"
              />
              {errors.secondaryAddress?.postcode?.message && (
                <InputError message={errors.secondaryAddress.postcode.message} />
              )}
            </InputWrap>
            <InputWrap>
              <InputGroup>
                <Select {...register('secondaryAddress.countryId')} defaultValue="GB">
                  <option key="default-option" value="">
                    None selected
                  </option>
                  {COUNTRY_OPTIONS.map(({ name, value }) => (
                    <option key={value} value={value}>
                      {name}
                    </option>
                  ))}
                </Select>
                <Label>Country</Label>
              </InputGroup>
            </InputWrap>
          </FormLayout>
          <div className={elMb7}>
            <Button onClick={handleAddAddress(setHasSecondaryAddress, hasSecondaryAddress)} intent="low">
              Remove Secondary Address
            </Button>
          </div>
        </>
      ) : (
        <div className={elMb7}>
          <Button onClick={handleAddAddress(setHasSecondaryAddress, hasSecondaryAddress)} intent="low">
            Add Secondary Address
          </Button>
        </div>
      )}
      {hasWorkAddress ? (
        <>
          <BodyText>Work Address</BodyText>
          <FormLayout className={cx(elMb11, elFadeIn)}>
            <InputWrap>
              <InputGroup>
                <Select {...register('workAddress.type')} defaultValue="work">
                  <option key="default-option" value="">
                    None selected
                  </option>
                  {ADDRESS_TYPE_OPTIONS.map(({ name, value }) => (
                    <option key={value} value={value}>
                      {name}
                    </option>
                  ))}
                </Select>
                <Label>Address Type</Label>
              </InputGroup>
            </InputWrap>
            <InputWrap>
              <InputGroup
                {...register('workAddress.buildingName')}
                type="text"
                label="Building Name"
                placeholder="Name of the building"
              />
              {errors.workAddress?.buildingName?.message && (
                <InputError message={errors.workAddress.buildingName.message} />
              )}
            </InputWrap>
            <InputWrap>
              <InputGroup
                {...register('workAddress.buildingNumber')}
                type="text"
                label="Building Number"
                placeholder="Number of the building"
              />
              {errors.workAddress?.buildingNumber?.message && (
                <InputError message={errors.workAddress.buildingNumber.message} />
              )}
            </InputWrap>
            <InputWrap>
              <InputGroup {...register('workAddress.line1')} type="text" label="Line 1" placeholder="Address line 1" />
              {errors.workAddress?.line1?.message && <InputError message={errors.workAddress.line1.message} />}
            </InputWrap>
            <InputWrap>
              <InputGroup {...register('workAddress.line2')} type="text" label="Line 2" placeholder="Address line 2" />
              {errors.workAddress?.line2?.message && <InputError message={errors.workAddress.line2.message} />}
            </InputWrap>
            <InputWrap>
              <InputGroup {...register('workAddress.line3')} type="text" label="Line 3" placeholder="Address line 3" />
              {errors.workAddress?.line3?.message && <InputError message={errors.workAddress.line3.message} />}
            </InputWrap>
            <InputWrap>
              <InputGroup {...register('workAddress.line4')} type="text" label="Line 4" placeholder="Address line 4" />
              {errors.workAddress?.line4?.message && <InputError message={errors.workAddress.line4.message} />}
            </InputWrap>
            <InputWrap>
              <InputGroup
                {...register('workAddress.postcode')}
                type="text"
                label="Post Code"
                placeholder="Address post code"
              />
              {errors.workAddress?.postcode?.message && <InputError message={errors.workAddress.postcode.message} />}
            </InputWrap>
            <InputWrap>
              <InputGroup>
                <Select {...register('workAddress.countryId')} defaultValue="GB">
                  <option key="default-option" value="">
                    None selected
                  </option>
                  {COUNTRY_OPTIONS.map(({ name, value }) => (
                    <option key={value} value={value}>
                      {name}
                    </option>
                  ))}
                </Select>
                <Label>Country</Label>
              </InputGroup>
            </InputWrap>
          </FormLayout>
          <div className={elMb7}>
            <Button onClick={handleAddAddress(setHasWorkAddress, hasWorkAddress)} intent="low">
              Remove Work Address
            </Button>
          </div>
        </>
      ) : (
        <div className={elMb7}>
          <Button onClick={handleAddAddress(setHasWorkAddress, hasWorkAddress)} intent="low">
            Add Work Address
          </Button>
        </div>
      )}
    </>
  )
}
