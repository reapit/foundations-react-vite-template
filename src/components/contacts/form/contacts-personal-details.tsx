import {
  BodyText,
  InputWrap,
  InputGroup,
  Label,
  Select,
  FormLayout,
  elFadeIn,
  elMb11,
  InputError,
} from '@reapit/elements'
import dayjs from 'dayjs'
import { FC } from 'react'
import { UseFormReturn } from 'react-hook-form'
import { ContactFormSchema } from './contacts-validation-schema'

export const TITLE_OPTIONS = ['Mr', 'Mrs', 'Ms', 'Miss', 'Mx', 'Dr', 'Prof']

interface ContactPersonalProps {
  form: UseFormReturn<ContactFormSchema, any>
}

export const ContactsPersonal: FC<ContactPersonalProps> = ({ form }) => {
  const {
    register,
    formState: { errors },
  } = form

  return (
    <>
      <div className={elMb11}>
        <BodyText hasGreyText hasNoMargin>
          Provide us with some basic personal information about your contact.
        </BodyText>
      </div>
      <FormLayout className={elFadeIn}>
        <InputWrap>
          <InputGroup>
            <Select {...register('title')}>
              <option key="default-option" value="">
                None selected
              </option>
              {TITLE_OPTIONS.map((option) => (
                <option key={option} value={option}>
                  {option}
                </option>
              ))}
            </Select>
            <Label>Please select a title</Label>
            {errors.title?.message && <InputError message={errors.title.message} />}
          </InputGroup>
        </InputWrap>
        <InputWrap>
          <InputGroup
            label="First Name"
            placeholder="Please enter a first name"
            {...register('forename')}
            icon={errors?.forename?.message ? 'asteriskSystem' : undefined}
            intent="danger"
          />
          {errors.forename?.message && <InputError message={errors.forename.message} />}
        </InputWrap>
        <InputWrap>
          <InputGroup
            label="Surname"
            placeholder="Please enter a surname"
            {...register('surname')}
            icon={errors?.surname?.message ? 'asteriskSystem' : undefined}
            intent="danger"
          />
          {errors.surname?.message && <InputError message={errors.surname.message} />}
        </InputWrap>
        <InputWrap>
          <InputGroup
            label="Date of birth"
            type="date"
            {...register('dateOfBirth')}
            max={dayjs().format('YYYY-MM-DD')}
            icon={errors?.dateOfBirth?.message ? 'asteriskSystem' : undefined}
            intent="danger"
          />
          {errors.dateOfBirth?.message && <InputError message={errors.dateOfBirth.message} />}
        </InputWrap>
      </FormLayout>
    </>
  )
}
