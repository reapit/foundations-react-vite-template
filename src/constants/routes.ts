import { CONTACTS_ROUTES } from '../components/contacts'
import { LOGIN_ROUTES } from '../components/login'

export const Routes = {
  HOME: '/',
  ...LOGIN_ROUTES,
  ...CONTACTS_ROUTES,
}
