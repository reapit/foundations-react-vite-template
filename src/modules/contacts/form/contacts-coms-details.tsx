import {
  BodyText,
  InputWrap,
  InputGroup,
  Label,
  Select,
  FormLayout,
  elFadeIn,
  elMb11,
  InputError,
  Toggle,
  ElToggleItem,
} from '@reapit/elements'
import { FC } from 'react'
import { UseFormReturn } from 'react-hook-form'
import { ContactFormSchema } from './contacts-validation-schema'

export const MARKETING_OPTIONS = [
  { value: 'grant', name: 'Granted' },
  { value: 'deny', name: 'Denied' },
  { value: 'notAsked', name: 'Not Asked' },
]

interface ContactsComsDetailsProps {
  form: UseFormReturn<ContactFormSchema, any>
}

export const ContactsComsDetails: FC<ContactsComsDetailsProps> = ({ form }) => {
  const {
    register,
    formState: { errors },
  } = form
  return (
    <>
      <div className={elMb11}>
        <BodyText hasGreyText hasNoMargin>
          Provide here the communication details and marketing preferences of your contact.
        </BodyText>
      </div>
      <FormLayout className={elFadeIn}>
        <InputWrap>
          <InputGroup
            label="Home Phone"
            placeholder="Please enter a telephone number"
            {...register('homePhone')}
            icon={errors?.homePhone?.message ? 'asteriskSystem' : undefined}
            intent="danger"
          />
          {errors.homePhone?.message && <InputError message={errors.homePhone.message} />}
        </InputWrap>
        <InputWrap>
          <InputGroup
            label="Work Phone"
            placeholder="Please enter a telephone number"
            {...register('workPhone')}
            icon={errors?.workPhone?.message ? 'asteriskSystem' : undefined}
            intent="danger"
          />
          {errors.workPhone?.message && <InputError message={errors.workPhone.message} />}
        </InputWrap>
        <InputWrap>
          <InputGroup
            label="Mobile Phone"
            placeholder="Please enter a telephone number"
            {...register('mobilePhone')}
            icon={errors?.mobilePhone?.message ? 'asteriskSystem' : undefined}
            intent="danger"
          />
          {errors.mobilePhone?.message && <InputError message={errors.mobilePhone.message} />}
        </InputWrap>
        <InputWrap>
          <InputGroup
            label="Email"
            type="email"
            placeholder="Please enter an email address"
            {...register('email')}
            icon={errors?.email?.message ? 'asteriskSystem' : undefined}
            intent="danger"
          />
          {errors.email?.message && <InputError message={errors.email.message} />}
        </InputWrap>
        <InputWrap>
          <InputGroup>
            <Select {...register('marketingConsent')}>
              <option key="default-option" value="">
                None selected
              </option>
              {MARKETING_OPTIONS.map(({ name, value }) => (
                <option key={value} value={value}>
                  {name}
                </option>
              ))}
            </Select>
            <Label>Marketing Consent</Label>
          </InputGroup>
          {errors.marketingConsent?.message && <InputError message={errors.marketingConsent.message} />}
        </InputWrap>
        <InputWrap>
          <InputGroup>
            <Toggle {...register('communicationPreferenceLetter')} id="com-pref-letter">
              <ElToggleItem>On</ElToggleItem>
              <ElToggleItem>Off</ElToggleItem>
            </Toggle>
            <Label>Communicate Preference Letter</Label>
          </InputGroup>
        </InputWrap>
        <InputWrap>
          <InputGroup>
            <Toggle {...register('communicationPreferenceEmail')} id="com-pref-email">
              <ElToggleItem>On</ElToggleItem>
              <ElToggleItem>Off</ElToggleItem>
            </Toggle>
            <Label>Communicate Preference Email</Label>
          </InputGroup>
        </InputWrap>
        <InputWrap>
          <InputGroup>
            <Toggle {...register('communicationPreferencePhone')} id="com-pref-phone">
              <ElToggleItem>On</ElToggleItem>
              <ElToggleItem>Off</ElToggleItem>
            </Toggle>
            <Label>Communicate Preference Phone</Label>
          </InputGroup>
        </InputWrap>
        <InputWrap>
          <InputGroup>
            <Toggle {...register('communicationPreferenceSMS')} id="com-pref-sms">
              <ElToggleItem>On</ElToggleItem>
              <ElToggleItem>Off</ElToggleItem>
            </Toggle>
            <Label>Communicate Preference SMS</Label>
          </InputGroup>
        </InputWrap>
      </FormLayout>
    </>
  )
}
