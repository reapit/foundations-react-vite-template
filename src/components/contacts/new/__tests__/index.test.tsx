import { ContactsNew, handleSwitchStep, handleSubmitContact } from '..'
import { render } from '../../../../scripts/tests'
import { ContactFormSchema } from '../../form/contacts-validation-schema'
import { CONTACTS_ROUTES } from '../../routes'

vi.mock('../../../../hooks', () => ({
  usePlatformGet: vi.fn(),
  usePlatformUpdate: vi.fn(() => [vi.fn()]),
}))

describe('ContactsNew', () => {
  it('should match a snapshot', () => {
    expect(render(<ContactsNew />).asFragment()).toMatchSnapshot()
  })
})

describe('handleSwitchStep', () => {
  it('should validate step 1', async () => {
    const step = '2'
    const selectedStep = '1'
    const trigger = vi.fn(() => Promise.resolve(true))
    const setSelectedStep = vi.fn()

    const curried = handleSwitchStep(step, selectedStep, trigger, setSelectedStep)

    curried()

    await Promise.resolve()

    expect(trigger).toHaveBeenCalledWith(['title', 'dateOfBirth', 'forename', 'surname'])
    expect(setSelectedStep).toHaveBeenCalledWith(step)
  })

  it('should validate step 2', async () => {
    const step = '3'
    const selectedStep = '2'
    const trigger = vi.fn(() => Promise.resolve(true))
    const setSelectedStep = vi.fn()

    const curried = handleSwitchStep(step, selectedStep, trigger, setSelectedStep)

    curried()

    await Promise.resolve()

    expect(trigger).toHaveBeenCalledWith([
      'homePhone',
      'workPhone',
      'mobilePhone',
      'email',
      'marketingConsent',
      'communicationPreferenceEmail',
      'communicationPreferencePhone',
      'communicationPreferenceLetter',
      'communicationPreferenceSMS',
    ])
    expect(setSelectedStep).toHaveBeenCalledWith(step)
  })

  it('should validate step 3', async () => {
    const step = '4'
    const selectedStep = '3'
    const trigger = vi.fn(() => Promise.resolve(true))
    const setSelectedStep = vi.fn()

    const curried = handleSwitchStep(step, selectedStep, trigger, setSelectedStep)

    curried()

    await Promise.resolve()

    expect(trigger).toHaveBeenCalledWith(['primaryAddress', 'secondaryAddress', 'workAddress'])
    expect(setSelectedStep).toHaveBeenCalledWith(step)
  })

  it('should validate step 4', async () => {
    const step = '1'
    const selectedStep = '4'
    const trigger = vi.fn(() => Promise.resolve(true))
    const setSelectedStep = vi.fn()

    const curried = handleSwitchStep(step, selectedStep, trigger, setSelectedStep)

    curried()

    await Promise.resolve()

    expect(trigger).toHaveBeenCalledWith(['negotiatorIds', 'categoryIds', 'officeIds', 'source'])
    expect(setSelectedStep).toHaveBeenCalledWith(step)
  })
})

describe('handleSubmitContact', () => {
  it('should submit a contact then navigate', async () => {
    window.location.pathname = '/personal'
    const updateContact = vi.fn(() => Promise.resolve(true))
    const navigate = vi.fn()
    const values = {
      surname: 'MOCK_NAME',
      categoryIds: 'MOCK_CATEGORY',
      negotiatorIds: 'MOCK_NEG',
      officeIds: 'MOCK_OFFICE',
      source: 'MOCK-SOURCE',
    } as ContactFormSchema

    const curried = handleSubmitContact(updateContact, navigate)

    await curried(values)

    expect(updateContact).toHaveBeenCalledWith({
      ...values,
      categoryIds: ['MOCK_CATEGORY'],
      negotiatorIds: ['MOCK_NEG'],
      officeIds: ['MOCK_OFFICE'],
      source: {
        id: 'MOCK',
        type: 'SOURCE',
      },
    })
    expect(navigate).toHaveBeenCalledWith(CONTACTS_ROUTES.CONTACTS_LIST)
  })
})
