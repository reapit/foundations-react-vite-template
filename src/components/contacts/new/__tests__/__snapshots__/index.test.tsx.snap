// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactsNew should match a snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="el-snack-holder"
      />
      <h1
        class="el-title"
      >
        Contacts New
      </h1>
      <form>
        <div
          class="el-steps-vertical"
        >
          <div
            class="el-step-vertical"
          >
            <div
              class="el-step-vertical-item"
            >
              <div
                class="el-is-active el-step"
                data-testid="step-0"
              >
                1
              </div>
            </div>
            <div
              class="el-step-vertical-content"
            >
              <div
                class="el-mb11"
              >
                <p
                  class="el-has-grey-text el-has-no-margin el-body-text"
                >
                  Provide us with some basic personal information about your contact.
                </p>
              </div>
              <div
                class="el-fade-in el-form-layout"
              >
                <div
                  class="el-input-wrap"
                >
                  <div
                    class="el-input-group"
                  >
                    <select
                      class="el-select"
                      name="title"
                    >
                      <option
                        value=""
                      >
                        None selected
                      </option>
                      <option
                        value="Mr"
                      >
                        Mr
                      </option>
                      <option
                        value="Mrs"
                      >
                        Mrs
                      </option>
                      <option
                        value="Ms"
                      >
                        Ms
                      </option>
                      <option
                        value="Miss"
                      >
                        Miss
                      </option>
                      <option
                        value="Mx"
                      >
                        Mx
                      </option>
                      <option
                        value="Dr"
                      >
                        Dr
                      </option>
                      <option
                        value="Prof"
                      >
                        Prof
                      </option>
                    </select>
                    <label
                      class="el-label"
                    >
                      Please select a title
                    </label>
                  </div>
                </div>
                <div
                  class="el-input-wrap"
                >
                  <div
                    class="el-input-group"
                  >
                    <input
                      class="el-input"
                      id="test-static-id"
                      name="forename"
                      placeholder="Please enter a first name"
                    />
                    <label
                      class="el-label"
                      for="test-static-id"
                    >
                      First Name
                    </label>
                  </div>
                </div>
                <div
                  class="el-input-wrap"
                >
                  <div
                    class="el-input-group"
                  >
                    <input
                      class="el-input"
                      id="test-static-id"
                      name="surname"
                      placeholder="Please enter a surname"
                    />
                    <label
                      class="el-label"
                      for="test-static-id"
                    >
                      Surname
                    </label>
                  </div>
                </div>
                <div
                  class="el-input-wrap"
                >
                  <div
                    class="el-input-group"
                  >
                    <input
                      class="el-input"
                      id="test-static-id"
                      max="2023-01-01"
                      name="dateOfBirth"
                      type="date"
                    />
                    <label
                      class="el-label"
                      for="test-static-id"
                    >
                      Date of birth
                    </label>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="el-mb11 el-ml8 el-button-group"
        >
          <div
            class="el-button-group-align-left el-button-group-inner"
          >
            <button
              class="el-intent-primary el-button"
              type="button"
            >
              <div
                class="el-button-loader"
              />
              Next
            </button>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <div
      class="el-snack-holder"
    />
    <h1
      class="el-title"
    >
      Contacts New
    </h1>
    <form>
      <div
        class="el-steps-vertical"
      >
        <div
          class="el-step-vertical"
        >
          <div
            class="el-step-vertical-item"
          >
            <div
              class="el-is-active el-step"
              data-testid="step-0"
            >
              1
            </div>
          </div>
          <div
            class="el-step-vertical-content"
          >
            <div
              class="el-mb11"
            >
              <p
                class="el-has-grey-text el-has-no-margin el-body-text"
              >
                Provide us with some basic personal information about your contact.
              </p>
            </div>
            <div
              class="el-fade-in el-form-layout"
            >
              <div
                class="el-input-wrap"
              >
                <div
                  class="el-input-group"
                >
                  <select
                    class="el-select"
                    name="title"
                  >
                    <option
                      value=""
                    >
                      None selected
                    </option>
                    <option
                      value="Mr"
                    >
                      Mr
                    </option>
                    <option
                      value="Mrs"
                    >
                      Mrs
                    </option>
                    <option
                      value="Ms"
                    >
                      Ms
                    </option>
                    <option
                      value="Miss"
                    >
                      Miss
                    </option>
                    <option
                      value="Mx"
                    >
                      Mx
                    </option>
                    <option
                      value="Dr"
                    >
                      Dr
                    </option>
                    <option
                      value="Prof"
                    >
                      Prof
                    </option>
                  </select>
                  <label
                    class="el-label"
                  >
                    Please select a title
                  </label>
                </div>
              </div>
              <div
                class="el-input-wrap"
              >
                <div
                  class="el-input-group"
                >
                  <input
                    class="el-input"
                    id="test-static-id"
                    name="forename"
                    placeholder="Please enter a first name"
                  />
                  <label
                    class="el-label"
                    for="test-static-id"
                  >
                    First Name
                  </label>
                </div>
              </div>
              <div
                class="el-input-wrap"
              >
                <div
                  class="el-input-group"
                >
                  <input
                    class="el-input"
                    id="test-static-id"
                    name="surname"
                    placeholder="Please enter a surname"
                  />
                  <label
                    class="el-label"
                    for="test-static-id"
                  >
                    Surname
                  </label>
                </div>
              </div>
              <div
                class="el-input-wrap"
              >
                <div
                  class="el-input-group"
                >
                  <input
                    class="el-input"
                    id="test-static-id"
                    max="2023-01-01"
                    name="dateOfBirth"
                    type="date"
                  />
                  <label
                    class="el-label"
                    for="test-static-id"
                  >
                    Date of birth
                  </label>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="el-mb11 el-ml8 el-button-group"
      >
        <div
          class="el-button-group-align-left el-button-group-inner"
        >
          <button
            class="el-intent-primary el-button"
            type="button"
          >
            <div
              class="el-button-loader"
            />
            Next
          </button>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
